lib/ide/parser/src/api.rs

```diff
@@ -1,20 +1,57 @@
 use crate::prelude::*;
+use ast::ID;
+use std::ops::Sub;
 
-pub type Ast = ast::Ast;
+use serde::Serialize;
+use serde::Deserialize;
 
+pub type Ast = ast::Ast;
 
 
 // ============
 // == Parser ==
 // ============
 
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct Index { pub value:usize }
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct Size { pub value:usize }
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct Span { pub index: Index, pub size: Size }
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct IDMap(pub Vec<(Span, ID)>);
```

Comment:
This one I think should live next to AST structures. While currently it lives in parser, we will soon want a method that generates this from AST and I don't really think it is specific to parser.